probe/endpoint/connection_tracker.go

```diff
@@ -0,0 +1,258 @@
+package endpoint
+
+import (
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/weaveworks/scope/probe/endpoint/procspy"
+	"github.com/weaveworks/scope/probe/process"
+	"github.com/weaveworks/scope/report"
+)
+
+// connectionTrackerConfig are the config options for the endpoint tracker.
+type connectionTrackerConfig struct {
+	HostID       string
+	HostName     string
+	SpyProcs     bool
+	UseConntrack bool
+	WalkProc     bool
+	UseEbpfConn  bool
+	ProcRoot     string
+	BufferSize   int
+	Scanner      procspy.ConnectionScanner
+	DNSSnooper   *DNSSnooper
+}
+
+type connectionTracker struct {
+	conf            connectionTrackerConfig
+	flowWalker      flowWalker // Interface
+	ebpfTracker     eventTracker
+	reverseResolver *reverseResolver
+}
+
+func newConnectionTracker(conf connectionTrackerConfig) connectionTracker {
+	if !conf.UseEbpfConn {
+		// ebpf OFF, use flowWalker
+		return connectionTracker{
+			conf:            conf,
+			flowWalker:      newConntrackFlowWalker(conf.UseConntrack, conf.ProcRoot, conf.BufferSize, "--any-nat"),
+			ebpfTracker:     nil,
+			reverseResolver: newReverseResolver(),
+		}
+	}
+	// When ebpf will be active by default, check if it starts correctly otherwise fallback to flowWalk
+	et, err := newEbpfTracker(conf.UseEbpfConn)
+	if err != nil {
+		// TODO: fallback to flowWalker, when ebpf is enabled by default
+		log.Errorf("Error setting up the ebpfTracker, connections will not be reported: %s", err)
+		noopConnectionTracker := connectionTracker{
+			conf:            conf,
+			flowWalker:      nil,
+			ebpfTracker:     nil,
+			reverseResolver: nil,
+		}
+		return noopConnectionTracker
+	}
+	// Run conntrack and proc parsing synchronously only once to initialize ebpfTracker
+	seenTuples := map[string]fourTuple{}
+	// Consult the flowWalker to get the initial state
+	if !conf.UseConntrack {
+		log.Warn("conntrack is disabled")
```

Comment:
Why do we need this? If the flag reached this point it should be intentional, shouldn't it?